### DISCUSSION TITLE
Failed to clone submodules

### DISCUSSION THREAD

**Discussion by fzCG2021** [codemagic-ci-cd/codemagic-docs#1500] [Q&A]
I have a Flutter app that was successfully publishing for both iOS and android until last 4days.
I also have a submodule inside my main branch, that was always there.
The question is: Is there any update in "Fetching app sources" that is causing it to fail?

Thanks

**Comment by olivierkorner** [comment]
Same issue here: yesterday, the cloning of submodule was failing in a native iOS app.
Today, the cloning of the submodule works, but the build times out while fetching the SPM dependencies of the project!

**Reply by icarusdust** [reply]
Hi @fzCG2021 @olivierkorner sorry for the inconvenience. The fix is live, please test it and let me know.

**Reply by olivierkorner** [reply]
Sorry, no difference, @icarusdust !
Stuck at: `Checking out ...` one of the dependencies.

**Reply by icarusdust** [reply]
Provide a build id please.

**Reply by olivierkorner** [reply]
6419ae9d3aa7b0610ba0f100

**Reply by icarusdust** [reply]
Thanks. So the submodule issue is fixed obviously and this is not related to it. I will look into it further and get back to you. 

On a different note, you are on a paid plan which means that you have access to the private chat widget in the UI, you should use that for your future queries. The reason it is better because our support team prioritises that channel.

**Reply by olivierkorner** [reply]
I couldn't see the widget, most likely because of the AdGuard extension…
Do I need to to contact the support team through the widget as well?

**Reply by icarusdust** [reply]
Great that you figured it out.

We can continue here this time, let's use the widget for future queries.

**Comment by fzCG2021** [comment]
Thanks @icarusdust for the fix.
It worked!
